File: gastoclaro_app/lib/models/debt.dart
class Debt {
  final int id;
  final String debtType;
  final String name;
  final String? creditorName;
  final String currency;
  final double? originalAmount;
  final double currentBalance;
  final double? monthlyDueAmount;
  final double? minimumPayment;
  final double? interestRateMonthly;
  final int? dueDay;
  final String status;
  final bool hasFixedPayment;
  final String? notes;

  Debt({
    required this.id,
    required this.debtType,
    required this.name,
    required this.creditorName,
    required this.currency,
    required this.originalAmount,
    required this.currentBalance,
    required this.monthlyDueAmount,
    required this.minimumPayment,
    required this.interestRateMonthly,
    required this.dueDay,
    required this.status,
    required this.hasFixedPayment,
    required this.notes,
  });

  factory Debt.fromJson(Map<String, dynamic> json) {
    double? toNullableDouble(dynamic value) {
      if (value == null) {
        return null;
      }

      if (value is int) {
        return value.toDouble();
      }

      if (value is double) {
        return value;
      }

      return double.tryParse(value.toString());
    }

    int? toNullableInt(dynamic value) {
      if (value == null) {
        return null;
      }

      if (value is int) {
        return value;
      }

      return int.tryParse(value.toString());
    }

    return Debt(
      id: toNullableInt(json['id']) ?? 0,
      debtType: json['debt_type']?.toString() ?? '',
      name: json['name']?.toString() ?? '',
      creditorName: json['creditor_name']?.toString(),
      currency: json['currency']?.toString() ?? 'PEN',
      originalAmount: toNullableDouble(json['original_amount']),
      currentBalance: toNullableDouble(json['current_balance']) ?? 0,
      monthlyDueAmount: toNullableDouble(json['monthly_due_amount']),
      minimumPayment: toNullableDouble(json['minimum_payment']),
      interestRateMonthly: toNullableDouble(json['interest_rate_monthly']),
      dueDay: toNullableInt(json['due_day']),
      status: json['status']?.toString() ?? 'active',
      hasFixedPayment: json['has_fixed_payment'] == true,
      notes: json['notes']?.toString(),
    );
  }
}

File: gastoclaro-api/app/Models/Debt.php
<?php

declare(strict_types=1);

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Debt extends Model
{
    protected $fillable = [
        'user_id',
        'debt_type',
        'name',
        'creditor_name',
        'currency',
        'original_amount',
        'current_balance',
        'monthly_due_amount',
        'minimum_payment',
        'interest_rate_monthly',
        'due_day',
        'status',
        'has_fixed_payment',
        'notes',
    ];

    protected function casts(): array
    {
        return [
            'original_amount' => 'decimal:2',
            'current_balance' => 'decimal:2',
            'monthly_due_amount' => 'decimal:2',
            'minimum_payment' => 'decimal:2',
            'interest_rate_monthly' => 'decimal:2',
            'has_fixed_payment' => 'boolean',
            'due_day' => 'integer',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}
